src/Runner.Common/ConfigurationStore.cs

```diff
@@ -15,6 +15,9 @@ namespace GitHub.Runner.Common
     [DataContract]
     public sealed class RunnerSettings
     {
+        [DataMember(Name = "IsHostedServer")]
```

Comment:
@TingluoHuang  let me know what you think about this approach. I left the PR in draft for now... looking for early feedback

Reply:
i am wonder whether we should just fix the system.servertype variable since it use to set to `onprem` or `hosted`?

Reply:
Setting up GHES now to test. Otherwise will only persist this setting for on-prem, so we can drop it later after alpha.